Expensify/App
# Beta race fixes

### Details


### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/171345

## Setup
1. We'll want this to behave just like stag/prod so there are some setup steps.
2. Remove isDevelopment from [here](https://github.com/Expensify/App/blob/c00918712a050e92166954462c7b309164277d4e/src/libs/Permissions.js#L11). That line returns true really fast with isDevelopment so you can't test race conditions.
3. Go to https://www.expensify.com.dev/_support/beta and add a test account to freePlan.
4. Delete this [block](https://github.com/Expensify/Web-Expensify/blob/84450ba039e56ab330e69aac7a252f7f3fe645f3/lib/BetaManager.php#L194-L199).
5. If you really want to stretch the race condition update this [line](https://github.com/Expensify/App/blob/f8c684f16087a831d7c85899e2370d2d21ea316f/src/libs/actions/User.js#L56) with,
```
setTimeout(() => Onyx.set(ONYXKEYS.BETAS, response.betas), 5000);
```
which is what I did while working on the code.

### Tests
Testing workspace/new
1. Go to Expensify App. While signed out visit http://localhost:8080/workspace/new.
2. Enter your credentials.
3. Once logged in confirm you see the workplace modal
![image](https://user-images.githubusercontent.com/14356145/126849671-8cfeeb58-463d-4a98-ae49-e1e990859e81.png)

Testing pricing redirect.
1. Logout of the Expensify App.
2. Go to expensify.com.dev. Log into your test account then go to https://www.expensify.com.dev/pricing
3. Select Free and confirm you see the workplace modal again
![image](https://user-images.githubusercontent.com/14356145/126849736-95133b51-985a-4ce8-9bd2-d25dc66e6d1f.png)


### QA Steps
Same steps as above except that you'll need to use an email that's on free plan beta.

### Tested On

### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
